ui/src/pages/version/components/forms/components/EnvironmentPanel.js

```diff
@@ -0,0 +1,142 @@
+import React, { Fragment, useContext } from "react";
+import {
+  EuiForm,
+  EuiFormRow,
+  EuiIcon,
+  EuiSuperSelect,
+  EuiText,
+  EuiTextColor,
+  EuiToolTip
+} from "@elastic/eui";
+import sortBy from "lodash/sortBy";
+import { Panel } from "./Panel";
+import EnvironmentsContext from "../../../../../providers/environments/context";
+
+const EnvironmentDropdownOption = ({
+  environment,
+  version,
+  endpoint,
+  disabled
+}) => {
+  const option = (
+    <Fragment>
+      <strong>{environment.name}</strong>
+      <EuiText size="s">
+        <EuiTextColor color="subdued">{environment.cluster}</EuiTextColor>
+      </EuiText>
+      {endpoint && (
+        <EuiText size="s">
+          <EuiTextColor color="subdued">{endpoint.status}</EuiTextColor>
+        </EuiText>
+      )}
+    </Fragment>
+  );
+
+  return disabled ? (
+    <EuiToolTip
+      position="left"
+      content={
+        <Fragment>
+          <p>
+            Model version {version.id} already has a <b>{endpoint.status}</b>{" "}
+            endpoint on <b>{environment.name}</b>.
+          </p>
+          <p>
+            <br />
+          </p>
+          <p>
+            If you want to do redeployment, go to model version details page and
+            click Redeploy button.
+          </p>
+        </Fragment>
+      }>
+      {option}
+    </EuiToolTip>
+  ) : (
+    option
+  );
+};
+
+const isOptionDisabled = endpoint => {
+  return (
+    endpoint &&
+    (endpoint.status === "serving" ||
+      endpoint.status === "running" ||
+      endpoint.status === "pending")
+  );
+};
+
+export const EnvironmentPanel = ({
+  environment,
+  version,
+  onChange,
+  errors = {},
+  isEnvironmentDisabled = false
+}) => {
+  const environments = useContext(EnvironmentsContext);
+
+  const environmentOptions = sortBy(environments, "name").map(environment => {
+    const versionEndpoint = version.endpoints.find(ve => {
+      return ve.environment_name === environment.name;
+    });
+
+    const isDisabled = isOptionDisabled(versionEndpoint);
+    return {
+      value: environment.name,
+      disabled: isDisabled,
+      inputDisplay: environment.name,
+      dropdownDisplay: (
+        <EnvironmentDropdownOption
+          environment={environment}
+          version={version}
+          endpoint={versionEndpoint}
+          disabled={isDisabled}
+        />
+      )
+    };
+  });
+
+  const onEnvironmentChange = value => {
+    onChange("environment_name")(value);
+
+    const existingEndpoint = version.endpoints.find(
```

Comment:
looks like the implementation is same with this one https://github.com/gojek/merlin/pull/127/files#diff-e778faffeaf52b967cad0c29be39d8c0027eabad18591a7d970860e97b1e8e16R79. Can you move it to separate function?